rasa/shared/importers/importer.py

```diff
@@ -196,6 +196,10 @@ def _importer_from_dict(
             config_path, domain_path, training_data_paths, **constructor_arguments
         )
 
+    def fingerprint(self) -> Text:
+        """Returns a static fingerprint."""
+        return "TrainingDataImporter"
```

Comment:
I had that at first - but I found it difficult to predict what the result of instantiating a TrainingDataImporter finally is ...

Reply:
@wochinge any thoughts on how we should fingerprint this?

Reply:
Do we actually need to? Looking at our diagram [here](https://www.notion.so/rasa/Rasa-Open-Source-3-0-Architecture-Implementation-Proposal-51ab90b05c41435ca98189a101676a1e#299b6801bc5842638f7915a1988f650b) I think it would be fine to have an arbitrary fingerprint (= always run this). The subsequent nodes (`TrainingDataProvider`, `DomainProvider`, `StoryGraphProvider`, etc.) will guarantee fingerprint-ability, won't they? 

The alternative would otherwise be to find all eligible files and get the hashes of the file content?

Reply:
Yup I thought the same about having arbitrary fingerprint (https://github.com/RasaHQ/rasa/issues/9132#issuecomment-883406385). I was just thinking that having class name would mean it was at least dependant on the subclass being used. However thinking a bit more now, if all it's children are input nodes it makes no difference what-so-ever right?

Reply:
I think it should be arbitrary because we otherwise might skip the schema validation / finetuning validation although the actual training data might have changed 🤔

Reply:
oh I see what you mean by arbitrary. But then we _always_ run validation. We originally didn't want to do that, hence not marking them input nodes right? In which case we would need a proper fingerprint of the files then.

Reply:
We are not always running it unless it's a target node, right?

Reply:
ah yes sorry - I mean always run validations during the fingerprint run. But I guess that's fine (I think we agreed it was fine).

Reply:
So do I understand correctly that the fingerprint of the TrainingDataImporter should be random? :)

Reply:
@joejuzl @wochinge ^

Reply:
👍🏻